### PR TITLE
Migrate license, collections and common modules test to junit 5

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

File: community/collections/src/test/java/org/neo4j/collection/PrimitiveLongArrayQueueTest.java [hunk 1]
```diff
@@ -204,10 +205,10 @@ public void addAllElementsFromOtherQueue()
         assertTrue( queue.isEmpty() );
     }
 
-    @Test( expected = AssertionError.class )
-    public void doNotAllowCreationOfQueueWithRandomCapacity()
+    @Test
+    void doNotAllowCreationOfQueueWithRandomCapacity()
     {
-        new PrimitiveLongArrayQueue( 7 );
+        assertThrows( AssertionError.class, () -> new PrimitiveLongArrayQueue( 7 ) );
```
Comment (andreikoval) [inline community/collections/src/test/java/org/neo4j/collection/PrimitiveLongArrayQueueTest.java]: eliminate assert statement in ctor and replace it with IAE

File: community/collections/src/test/java/org/neo4j/collection/pool/LinkedQueuePoolTest.java [hunk 2]
```diff
@@ -339,11 +338,11 @@ public void release()
 
     private static class StatefulMonitor implements LinkedQueuePool.Monitor<Object>
     {
-        public AtomicInteger currentPeakSize = new AtomicInteger( -1 );
-        public AtomicInteger targetSize = new AtomicInteger( -1 );
+        AtomicInteger currentPeakSize = new AtomicInteger( -1 );
+        AtomicInteger targetSize = new AtomicInteger( -1 );
         public AtomicInteger created = new AtomicInteger( 0 );
```
Comment (andreikoval) [inline community/collections/src/test/java/org/neo4j/collection/pool/LinkedQueuePoolTest.java]: do these still need to be public?

File: community/collections/src/test/java/org/neo4j/helpers/collection/TestCommonIterators.java [hunk 3]
```diff
@@ -55,175 +56,175 @@ public void testNoDuplicatesFilteringIterator()
...
 
         // first Iterator
-        assertEquals( object, Iterators.first( Arrays.asList( object, object2 ).iterator() ) );
-        assertEquals( object, Iterators.first( Arrays.asList( object ).iterator() ) );
+        assertEquals( object, Iterators.first( asList( object, object2 ).iterator() ) );
+        assertEquals( object, Iterators.first( asList( object ).iterator() ) );
         try
         {
-            Iterators.first( Arrays.asList().iterator() );
+            Iterators.first( asList().iterator() );
```
Comment (andreikoval) [inline community/collections/src/test/java/org/neo4j/helpers/collection/TestCommonIterators.java]: for consistency pls use `assertThrows` here and in other places

File: community/common/src/test/java/org/neo4j/ports/allocation/PortRepositoryIT.java [hunk 4]
```diff
@@ -69,8 +71,8 @@ public void shouldCoordinateUsingFileSystem() throws Exception
     }
 
     @Test
-    @Ignore
-    public void shouldNotInterfereWithOtherRepositories() throws Exception
+    @Disabled
```
Comment (andreikoval) [inline community/common/src/test/java/org/neo4j/ports/allocation/PortRepositoryIT.java]: maybe delete it?

**Review by andreikoval — CHANGES_REQUESTED** [review]

**Comment by MishaDemianenko** [comment]
@andreikoval here we go. All comments should be fixed now.

File: community/collections/src/main/java/org/neo4j/collection/PrimitiveLongArrayQueue.java [hunk 5]
```diff
@@ -45,7 +45,10 @@ public PrimitiveLongArrayQueue()
 
     PrimitiveLongArrayQueue( int capacity )
     {
-        assert (capacity != 0) && ((capacity & (capacity - 1)) == 0) : "Capacity should be power of 2.";
+        if ( capacity == 0 || (capacity & (capacity - 1)) != 0 )
+        {
+            throw new IllegalArgumentException( "Capacity should be power of 2." );
```
Comment (andreikoval) [inline community/collections/src/main/java/org/neo4j/collection/PrimitiveLongArrayQueue.java]: maybe add actual value to the error message?

**Review by andreikoval — APPROVED** [review]

File: community/collections/src/main/java/org/neo4j/collection/PrimitiveLongArrayQueue.java (hunk 5 above)
Comment (MishaDemianenko) [reply community/collections/src/main/java/org/neo4j/collection/PrimitiveLongArrayQueue.java]: yes updated

**Pull request merged**